server/rest/wms_dataset.py

```diff
@@ -34,12 +36,28 @@ def __init__(self):
     @loadmodel(map={'wmsSourceId': 'wmsSource'}, model='item',
                level=AccessType.READ)
     def createWmsDataset(self, wmsSource, params):
+        # Get layer legend (TODO// Include authentication in the future)
+        # Legend to be included in the metadata?
+        base_url = wmsSource['meta']['minerva']['wms_params']['base_url']
+        hostName = base_url[7:28]
```

Comment:
call [getUrlParts](https://github.com/girder/girder/blob/master/girder/api/rest.py#L39-L57)

your current method is more brittle, and wouldn't give you what you expect for e.g. https